src/frontend/src/optimizer/plan_node/mod.rs

```diff
@@ -450,6 +451,14 @@ impl PlanRef {
                     })
                     .reduce(|a, b| a.or(b))
                     .unwrap();
+
+                if merge_predicate.conjunctions.len() == 1 {
+                    let mut expr_rewriter = ExpressionSimplifyRewriter {};
+                    merge_predicate = Condition::with_expr(
+                        expr_rewriter.rewrite_cond(merge_predicate.conjunctions[0].clone()),
+                    );
+                }
```

Comment:
Why only apply this rewrite to predicate with conjunction.len() = 1?

Reply:
thanks for pointing this out, indeed it could rewrite the entire conjunction.